server/hotrod/src/test/java/org/infinispan/server/hotrod/HotRodMergeTest.java

```diff
@@ -92,12 +94,15 @@ public void testNewTopologySentAfterOverlappingMerge() {
       int initialTopology = advancedCache(0).getRpcManager().getTopologyId();
       expectCompleteTopology(client, initialTopology);
       PartitionDescriptor p1 = new PartitionDescriptor(0);
+      // isolatePartitions will always result in a CR fail as Node 0 tries to contact Node 1 in order to receive segments
+      // which is not possible as all messages received by Node 1 from Node 0 are discarded by the DISCARD protocol.
+      // Therefore, it is necessary for the state transfer timeout to be < then the timeout utilised by TestingUtil::waitForNoRebalance
```

Comment:
Maybe I'm not reading this right, but it sounds like a serious limitation of conflict resolution.

If the cluster splits, each partition may start its own conflict resolution, and it should only wait for nodes that are still in the same partition (i.e. in the same JGroups cluster view).